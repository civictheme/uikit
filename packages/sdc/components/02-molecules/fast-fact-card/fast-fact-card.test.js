import DrupalAttribute from 'drupal-attribute';

const template = 'components/02-molecules/fast-fact-card/fast-fact-card.twig';

describe('Fast Fact Card Component', () => {
  test('renders with required attributes', async () => {
    const c = await dom(template, {
      title: 'Fast Fact Title',
      summary: 'This is a summary of the fast fact.',
    });

    const element = c.querySelector('.ct-fast-fact-card');
    expect(c.querySelectorAll('.ct-fast-fact-card')).toHaveLength(1);
    expect(element.classList.contains('ct-theme-light')).toBe(true);
    expect(c.querySelector('.ct-fast-fact-card__title').textContent.trim()).toEqual('Fast Fact Title');
    expect(c.querySelector('.ct-fast-fact-card__summary')).toBeTruthy();

    const image = c.querySelector('.ct-fast-fact-card__image');
    expect(image).toBeNull();

    assertUniqueCssClasses(c);
  });

  test('renders with optional attributes', async () => {
    const c = await dom(template, {
      image: { url: 'https://example.com/image.jpg', alt: 'Image description' },
      title: 'Fast Fact Title',
      summary: 'This is a summary of the fast fact.',
      link: { text: 'Learn more', url: 'https://example.com', is_new_window: true },
      theme: 'dark',
      attributes: new DrupalAttribute().setAttribute('data-test', 'true'),
      modifier_class: 'custom-class',
    });

    const element = c.querySelector('.ct-fast-fact-card');
    expect(element).not.toBeNull();
    expect(element.classList.contains('ct-theme-dark')).toBe(true);
    expect(element.classList.contains('ct-fast-fact-card--card-clickable')).toBe(true);
    expect(element.classList.contains('custom-class')).toBe(true);
    expect(element.getAttribute('data-test')).toEqual('true');

    expect(c.querySelector('.ct-fast-fact-card__title').textContent.trim()).toContain('Fast Fact Title');
    expect(c.querySelector('.ct-fast-fact-card__summary')).toBeTruthy();

    const image = c.querySelector('.ct-fast-fact-card__image img');
    expect(image).toBeTruthy();
    expect(image.getAttribute('src')).toEqual('https://example.com/image.jpg');
    expect(image.getAttribute('alt')).toEqual('Image description');

    const titleLink = c.querySelector('.ct-fast-fact-card__title-link');
    expect(titleLink).toBeTruthy();
    expect(titleLink.getAttribute('href')).toEqual('https://example.com');
    expect(titleLink.getAttribute('target')).toEqual('_blank');

    assertUniqueCssClasses(c);
  });

  test('renders with link when provided', async () => {
    const c = await dom(template, {
      title: 'Fast Fact Title',
      summary: 'This is a summary of the fast fact.',
      link: { text: 'Learn more', url: 'https://example.com' },
    });

    const link = c.querySelector('.ct-fast-fact-card__title-link');
    expect(link).toBeTruthy();
    expect(link.getAttribute('href')).toEqual('https://example.com');

    assertUniqueCssClasses(c);
  });

  test('renders with title click only when is_title_click is true', async () => {
    const c = await dom(template, {
      title: 'Fast Fact Title',
      summary: 'This is a summary of the fast fact.',
      link: { text: 'Learn more', url: 'https://example.com' },
      is_title_click: true,
    });

    const element = c.querySelector('.ct-fast-fact-card');
    expect(element).not.toBeNull();
    expect(element.classList.contains('ct-fast-fact-card--card-clickable')).toBe(false);

    const titleLink = c.querySelector('.ct-fast-fact-card__title-link');
    expect(titleLink).toBeTruthy();
    expect(titleLink.getAttribute('href')).toEqual('https://example.com');

    assertUniqueCssClasses(c);
  });
});
