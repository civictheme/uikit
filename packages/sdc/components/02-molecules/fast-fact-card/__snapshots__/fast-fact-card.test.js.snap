// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fast Fact Card Component renders with link when provided 1`] = `
<div>
  



  <div
    class="ct-fast-fact-card ct-theme-light ct-fast-fact-card--card-clickable "
  >
    
        
            
    <div
      class="ct-fast-fact-card__title"
    >
      
                  
      <a
        class="ct-fast-fact-card__title-link"
        href="https://example.com"
      >
        Fast Fact Title
      </a>
      
              
    </div>
    
      
            


    <div
      class="ct-paragraph ct-theme-light ct-paragraph--large  ct-fast-fact-card__summary"
    >
      
    This is a summary of the fast fact.  
    </div>
          
  </div>
  

</div>
`;

exports[`Fast Fact Card Component renders with optional attributes 1`] = `
<div>
  



  <div
    class="ct-fast-fact-card ct-theme-dark ct-fast-fact-card--card-clickable custom-class"
    data-test="true"
  >
    
            
    <div
      class="ct-fast-fact-card__image"
    >
      
        

  
      <img
        alt="Image description"
        class="ct-image ct-theme-dark "
        src="https://example.com/image.jpg"
      />
      
      
    </div>
    
      
            
    <div
      class="ct-fast-fact-card__title"
    >
      
                  
      <a
        class="ct-fast-fact-card__title-link"
        href="https://example.com"
        target="_blank"
      >
        Fast Fact Title
      </a>
      
              
    </div>
    
      
            


    <div
      class="ct-paragraph ct-theme-dark ct-paragraph--large  ct-fast-fact-card__summary"
    >
      
    This is a summary of the fast fact.  
    </div>
          
  </div>
  

</div>
`;

exports[`Fast Fact Card Component renders with required attributes 1`] = `
<div>
  



  <div
    class="ct-fast-fact-card ct-theme-light  "
  >
    
        
            
    <div
      class="ct-fast-fact-card__title"
    >
      
                  Fast Fact Title
              
    </div>
    
      
            


    <div
      class="ct-paragraph ct-theme-light ct-paragraph--large  ct-fast-fact-card__summary"
    >
      
    This is a summary of the fast fact.  
    </div>
          
  </div>
  

</div>
`;

exports[`Fast Fact Card Component renders with title click only when is_title_click is true 1`] = `
<div>
  



  <div
    class="ct-fast-fact-card ct-theme-light  "
  >
    
        
            
    <div
      class="ct-fast-fact-card__title"
    >
      
                  
      <a
        class="ct-fast-fact-card__title-link"
        href="https://example.com"
      >
        Fast Fact Title
      </a>
      
              
    </div>
    
      
            


    <div
      class="ct-paragraph ct-theme-light ct-paragraph--large  ct-fast-fact-card__summary"
    >
      
    This is a summary of the fast fact.  
    </div>
          
  </div>
  

</div>
`;
